tests/ignite/metrics/test_accuracy.py

```diff
@@ -91,145 +91,45 @@ def _test():
         _test()
 
 
-def test_binary_input_NL():
-    # Binary accuracy on input of shape (N, L)
-    def _test():
-        acc = Accuracy()
-
-        y_pred = torch.randint(0, 2, size=(10, 5)).long()
-        y = torch.randint(0, 2, size=(10, 5)).long()
-        acc.update((y_pred, y))
-        np_y = y.numpy().ravel()
-        np_y_pred = y_pred.numpy().ravel()
-        assert acc._type == "binary"
-        assert isinstance(acc.compute(), float)
-        assert accuracy_score(np_y, np_y_pred) == pytest.approx(acc.compute())
-
-        acc.reset()
-        y_pred = torch.randint(0, 2, size=(10, 1, 5)).long()
-        y = torch.randint(0, 2, size=(10, 1, 5)).long()
-        acc.update((y_pred, y))
-        np_y = y.numpy().ravel()
-        np_y_pred = y_pred.numpy().ravel()
-        assert acc._type == "binary"
-        assert isinstance(acc.compute(), float)
-        assert accuracy_score(np_y, np_y_pred) == pytest.approx(acc.compute())
-
-        # Batched Updates
-        acc.reset()
-        y_pred = torch.randint(0, 2, size=(100, 8)).long()
-        y = torch.randint(0, 2, size=(100, 8)).long()
-
-        n_iters = 16
-        batch_size = y.shape[0] // n_iters + 1
-
-        for i in range(n_iters):
-            idx = i * batch_size
-            acc.update((y_pred[idx : idx + batch_size], y[idx : idx + batch_size]))
-
-        np_y = y.numpy().ravel()
-        np_y_pred = y_pred.numpy().ravel()
-        assert acc._type == "binary"
-        assert isinstance(acc.compute(), float)
-        assert accuracy_score(np_y, np_y_pred) == pytest.approx(acc.compute())
-
-    # check multiple random inputs as random exact occurencies are rare
-    for _ in range(10):
-        _test()
-
-
-def test_binary_input_NHW():
-    # Binary accuracy on input of shape (N, H, W, ...)
-    def _test():
+@pytest.mark.parametrize(
+    "y_pred,y,n_iters",
+    [
+        # Binary accuracy on input of shape (N, L)
+        (torch.randint(0, 2, size=(10, 5)).long(), torch.randint(0, 2, size=(10, 5)).long(), 1),
```

Comment:
Nevertheless, test code is now more compact it does not do the same thing as previously.
We explicitly call `reset` to check that the same `acc` instance can be reused.

Reply:
I just checked the accuracy method
```      def reset(self) -> None:
        self._num_correct = torch.tensor(0, device=self._device)
        self._num_examples = 0
        super(Accuracy, self).reset()``
So what it does is that it just resets the two variables that were updated in the `update` method.
When we use the parametrize method it creates a new instance which have the same effect. so In that case it won't affect the behavior of the test/instance. What do you think?

Reply:
That's a good point. I think for this case - even for the accuracy one - we can have dedicated test for the reset method i.e. `test_reset` that will make two consecutive calls with reset in between to make sure we actually reset as expected. I feel it's a bit of too much overhead when we do that in every single test case we are having. I am up for both solutions though

Reply:
Probably, depending on the metric, `test_reset` can have as much case as other tests... 
When we explicitly call `reset`, it emulates the usage of the metric inside evaluator.